src/semeio/workflows/ahm_analysis/ahmanalysis.py

```diff
@@ -164,18 +171,24 @@ def run(
         # loop over keys and calculate the KS matrix,
         # conditioning one parameter at the time.
         field_output = {}
+        updated_combinations = deepcopy(combinations)
         for group_name, obs_group in combinations.items():
             print("Processing:", group_name)
 
             #  Use localization to evaluate change of parameters for each observation
-            with tempfile.TemporaryDirectory():
+            with tempfile.TemporaryDirectory(), open_storage("storage", "w") as storage:
```

Comment:
Maybe deserves a comment? Something like this
```suggestion
            # The ordering of the context managers matters, we want to open a new storage in the
            # temporary directory
            with tempfile.TemporaryDirectory(), open_storage("storage", "w") as storage:
```

Reply:
RIP formatting...